include/nlohmann/detail/conversions/from_json.hpp

```diff
@@ -144,16 +143,20 @@ void from_json(const BasicJsonType& j, std::forward_list<T, Allocator>& l)
 }
 
 // valarray doesn't have an insert method
-template<typename BasicJsonType, typename T,
-         enable_if_t<std::is_convertible<BasicJsonType, T>::value, int> = 0>
-void from_json(const BasicJsonType& j, std::valarray<T>& l)
+template<typename BasicJsonType, typename T>
+auto from_json(const BasicJsonType& j, std::valarray<T>& l) -> decltype(j.template get<T>(), void())
 {
     if (JSON_HEDLEY_UNLIKELY(!j.is_array()))
     {
         JSON_THROW(type_error::create(302, "type must be array, but is " + std::string(j.type_name())));
     }
     l.resize(j.size());
-    std::copy(j.begin(), j.end(), std::begin(l));
+    std::transform(
```

Comment:
What's the benefit of using transform rather than copy?

Reply:
copy requires implicit conversions